resim/msg/pose.proto

```diff
@@ -0,0 +1,30 @@
+// Copyright 2023 ReSim, Inc.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+syntax = "proto3";
+
+import "resim/transforms/proto/se3.proto";
+
+package resim.msg;
+
+message PoseWithCovariance {
+    resim.transforms.proto.SE3 pose = 1;
+
+    // This is a 6x6 matrix in row-major order
+    repeated double covariance = 2;
+}
+
+message Twist {
+    // This has six elements with angular components first and linear second.
+    repeated double algebra = 1;
+}
+
+message TwistWithCovariance {
+    Twist twist = 1;
+
+    // This is a 6x6 matrix in row-major order
+    repeated double covariance = 2;
+}
```

Comment:
[nit] Perhaps slightly longer comments (or references to relevant parts of SE3) could be good here - mostly as it's very plausible this is the first place someone looks in our codebase!